bugbug/bug_features.py

```diff
@@ -224,6 +224,16 @@ def __call__(self, bug, **kwargs):
         return sum(commit['files_modified_num'] for commit in bug['commits'])
 
 
+class components_touched(object):
+    def __call__(self, bug, **kwargs):
+        return (ct for commit in bug['commits'] for ct in commit['components'])
+
+
+class components_touched_num(object):
+    def __call__(self, bug, **kwargs):
+        return sum(len(commit['components']) for commit in bug['commits'])
```

Comment:
We shouldn't sum all components, because there could be three patches touching exactly the same components.
So, you should first remove the duplicate components and then return the length of the result.

Reply:
So maybe we can use a set and update it with all the components first and then find the length of the set?

The same needs to be done in `components_touched` also right? So can we make a set there and then call it in `components_touched_num` to get the set and find its length?

Reply:
Yes, using a set would be good.